content/en/blog/2019/alb-ingress-gateway-iks/index.md

```diff
@@ -0,0 +1,394 @@
+having only Istio ingress gateway as your main entry point, you can continue to use the traditional Ingress for
+non-Istio services while using the Istio ingress gateway for services that are part of the mesh.
+
+IKS provides a convenient way for clients to access Istio ingress gateway by letting you
+[register a DNS domain](https://cloud.ibm.com/docs/containers?topic=containers-loadbalancer_hostname) for the gateway's
+IP with an IKS command. The domain is in the following [format](https://cloud.ibm.com/docs/containers?topic=containers-loadbalancer_hostname#loadbalancer_hostname_format): `<cluster_name>-<globally_unique_account_HASH>-0001.<region>.containers.appdomain.cloud`, for example `mycluster-a1b2cdef345678g9hi012j3kl4567890-0001.us-south.containers.appdomain.cloud`. In the same way as for the ALB domain, IKS provides a certificate and a private key,
+storing them in another Kubernetes secret.
+
+In this blog post you configure the IKS Ingress ALB to direct traffic to the services inside an Istio service mesh
+through the Istio ingress gateway, while using mutual TLS authentication between ALB and the gateway. For the mutual TLS authentication, you configure ALB and the Istio ingress gateway to use the certificates and the keys provided by IKS.
+Using certificates provided by IKS saves you the overhead of managing your own certificates for the connection between ALB and
+the Istio ingress gateway.
+
+The traffic to the services without Istio sidecar can continue to flow as before, directly from ALB.
+
+The diagram below exemplifies the described setting. It shows two services in the cluster, `service A` and `service B`.
+`service A` has Istio sidecar injected and it requires mutual TLS. `service B` has no Istio sidecar. `service B` can
+be accessed by clients through the ALB, while ALB directly communicates with `service B`. `service A` can be also
+accessed by clients through ALB, but in this case the traffic has to pass through the Istio ingress gateway. Mutual
+TLS authentication between ALB and the gateway is based on the certificates provided by IKS.
+The clients can also access the Istio ingress gateway directly. IKS registers different DNS domains for ALB and for the
+ingress gateway.
+
+{{< image width="100%" link="./alb-ingress-gateway.svg" caption="A cluster with ALB and Istio ingress gateway" >}}
+
+## Initial setting
+
+1.  Create the `httptools` namespace and enable Istio sidecar injection:
+
+    {{< text bash >}}
+    $ kubectl create namespace httptools
+    $ kubectl label namespace httptools istio-injection=enabled
+    namespace/httptools created
+    namespace/httptools labeled
+    {{< /text >}}
+
+1.  Deploy the `httpbin` sample to `httptools`:
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/httpbin/httpbin.yaml@ -n httptools
+    service/httpbin created
+    deployment.apps/httpbin created
+    {{< /text >}}
+
+1.  Enforce mutual TLS for the traffic to the services in
```

Comment:
Suggest to remove creating this Policy and DR for httptools
1. Keep blog focused on the ALB -> Gateway
2. Policy not used in 1.5
3. DR not required with automtls

Reply:
Fixed in https://github.com/istio/istio.io/pull/5098/commits/a99af4d702bece9f8749b89233366d345b5478b0. Enforcing mutual TLS was done to show why you need the traffic to go through the Istio Ingress Gateway - because of the mutual TLS. But, I guess, we can just explain it without actually enforcing it.